client/app/components/QueryBasedParameterInput.jsx

```diff
@@ -53,7 +55,13 @@ export class QueryBasedParameterInput extends React.Component {
             this.props.onSelect(options[0].value);
           }
         }
-      });
+      };
+
+      if (this.props.parentQueryId) {
```

Comment:
Gosh, at first it felt like a "d'oh! how didn't I do _this_ before?", but digging into this, I kinda feel that while putting this function on `Query` feels like a more natural place for this logic, it will probably look uglier because of the multiple layers of function signature that will have to carry this function across in order to get to `QueryBasedParameterInput`. (that is - `QueryBasedParameterInput` props, `ParameterValueInput` props, `parameter-value-input` directive, `parameters` directive and `query` page)

Let me know which you prefer.

Reply:
Isn't it the same amount of effort to pass around the Query object as passing around the parentQueryId? What's the difference?

Also, what happens on dashboard level parameters? We will pick one of the queries to serve as a parent? 🤔 

I wonder if there is another way to approach this.